src/test/java/networkbook/logic/parser/ParserUtilTest.java

```diff
@@ -150,6 +152,29 @@ public void parseEmail_validValueWithWhitespace_returnsTrimmedEmail() throws Exc
         assertEquals(expectedEmail, ParserUtil.parseEmail(emailWithWhitespace));
     }
 
+    @Test
+    public void parseEmails_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseEmails(null));
+    }
+
+    @Test
+    public void parseEmails_collectionWithInvalidEmails_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseEmails(Arrays.asList(VALID_EMAIL, INVALID_EMAIL)));
+    }
+
+    @Test
+    public void parseEmails_emptyCollection_returnsEmptyListOfEmails() throws Exception {
```

Comment:
I don't think we are supposed to let this function throw an Exception?

Reply:
Please take a look at how other test cases are constructed for `ParserUtilTest`. They all have to throw exception in order for Java to compile. It is the same thing as writing a try-catch block, but if the method throws an exception, the test case is automatically failed.